pkg/k8shandler/statefulset.go

```diff
@@ -448,43 +448,45 @@ func (node *statefulSetNode) isChanged() bool {
 		return false
 	}
 
-	// we will only have one container, no need to do range
-	nodeContainer := node.self.Spec.Template.Spec.Containers[0]
-	desiredContainer := desired.Spec.Template.Spec.Containers[0]
+	// Only Image and Resources (CPU & memory) differences trigger rolling restart
+	for index := 0; index < len(node.self.Spec.Template.Spec.Containers); index++ {
+		nodeContainer := node.self.Spec.Template.Spec.Containers[index]
```

Comment:
Do we need to ensure we are compairing the same container ? (i.e. name chech)

Reply:
they should be the same because that's the order we create them in...
but i could add that as a check..